auth/src/main/java/com/firebase/ui/auth/ui/accountmanagement/AccountSettingsActivity.java

```diff
@@ -0,0 +1,269 @@
+            if (row != null) {
+                providerImage = (ImageView) row.findViewById(R.id.provider_image);
+                providerName = (TextView) row.findViewById(R.id.provider_name);
+                displayName = (TextView) row.findViewById(R.id.display_name);
+                displayName.setText(user.getDisplayName());
+                ImageButton unlinkButton = (ImageButton) row.findViewById(R.id.unlink_idp_button);
+                unlinkButton.setOnClickListener(new UnlinkIdpClickListener(providerId));
+                int px;
+                switch (providerId) {
+                    case AuthUI.EMAIL_PROVIDER:
+                        providerImage.setBackground(makeColoredCircle(R.color.gdi_email_provider_grey));
+                        providerImage.setImageResource(R.drawable.ic_email_white_48dp);
+                        providerName.setText(R.string.idp_name_email);
+                        break;
+                    case AuthUI.FACEBOOK_PROVIDER:
+                        px = pxFromDp(10);
+                        providerImage.setPadding(px, px, px, px);
+                        providerImage.setBackground(makeColoredCircle(R.color.gdi_facebook_blue));
+                        providerImage.setImageResource(R.drawable.ic_facebook_white_22dp);
+                        providerName.setText(R.string.idp_name_facebook);
+                        break;
+                    case AuthUI.GOOGLE_PROVIDER:
+                        providerImage.setBackground(
+                                makeColoredCircle(R.color.gdi_white_background));
+                        providerImage.setImageResource(R.drawable.ic_googleg_color_24dp);
+                        providerName.setText(R.string.idp_name_google);
+                        break;
+                    case AuthUI.TWITTER_PROVIDER:
+                        providerImage.setBackground(makeColoredCircle(R.color.gdi_twitter_blue));
+                        providerImage.setImageResource(R.drawable.ic_twitter_bird_white_24dp);
+                        px = pxFromDp(10);
+                        providerImage.setPadding(px, px, px, px);
+                        providerName.setText(R.string.idp_name_twitter);
+                        break;
+                }
+
+            }
+        }
+    }
+
+    private Intent getReauthIntent() {
+        return AuthUI.getInstance().createReauthIntentBuilder()
+                .setTosUrl(mFlowParameters.termsOfServiceUrl)
+                .setLogo(mFlowParameters.logoId)
+                .setProviders(mFlowParameters.providerInfo)
+                .setIsSmartLockEnabled(mFlowParameters.smartLockEnabled)
+                .build();
+    }
+
+    private ShapeDrawable makeColoredCircle(@ColorRes int color) {
+        ShapeDrawable shapeDrawable = new ShapeDrawable();
+        shapeDrawable.setShape(new OvalShape());
+        int resolvedColor = getResources().getColor(color);
```

Comment:
I have some bad memories of getColor, we should use ContextCompat.getColor() I think (honestly can't remember why(